client/state/selectors/is-user-registration-days-within-range.js

```diff
@@ -1,35 +1,28 @@
-/** @format */
-
-/**
- * External dependencies
- */
-import moment from 'moment';
-
 /**
  * Internal dependencies
  */
 import { getCurrentUserDate } from 'state/current-user/selectors';
 
+const DAY_IN_MS = 1000 * 60 * 60 * 24;
+
 /**
  * Returns true if the number of days the current user has been registered for falls within the specied range of values.
  *
  * @param {Object} state Global state tree
- * @param {Object} momentInTime Current moment for determination of elapsed days since registration
+ * @param {Date} refDate Date for determination of elapsed days since registration
  * @param {Number} from Lower bound on days
  * @param {Number} to Upper bound on days
  * @return {?Boolean} True if the number of days falls withing the specified range
  */
-const isUserRegistrationDaysWithinRange = ( state, momentInTime, from, to ) => {
+export default function isUserRegistrationDaysWithinRange( state, refDate, from, to ) {
 	const date = getCurrentUserDate( state );
 
 	if ( ! date ) {
 		return null;
 	}
 
-	momentInTime = momentInTime || moment();
-	const days = momentInTime.diff( date, 'days', true );
+	refDate = refDate || Date.now();
+	const days = ( refDate - new Date( date ) ) / DAY_IN_MS;
```

Comment:
I was suspicious of this change at first, because I expected that a `moment.diff` of `2000-01-01T23:59:00` and `2000-01-02T00:00:00` would return `1`, but it actually returns `0` ( `0.0006944444444444445` here because we also pass `true` as the third param).

Reply:
Right I had to go digging around in the docs for what that `true` meant... This is why options objects are better 😄